builder/message_builder.go

```diff
@@ -3,18 +3,17 @@ package builder
 import (
 	"context"
 	"fmt"
-
 	"github.com/iotaledger/hive.go/serializer/v2"
 	"github.com/iotaledger/iota.go/v3"
 	"github.com/iotaledger/iota.go/v3/nodeclient"
 	"github.com/iotaledger/iota.go/v3/pow"
 )
 
 // NewMessageBuilder creates a new MessageBuilder.
-func NewMessageBuilder() *MessageBuilder {
+func NewMessageBuilder(protoVersion byte) *MessageBuilder {
 	return &MessageBuilder{
 		msg: &iotago.Message{
-			ProtocolVersion: iotago.ProtocolVersion,
+			ProtocolVersion: protoVersion,
```

Comment:
Is this change necessary?
The message builder should always have the same protocol version like the created message.

Reply:
Yes, because the library might support multiple versions at the same time, there isn't a single lib version per protocol version.